app/static/favorite_quotes.js

```diff
@@ -0,0 +1,48 @@
+// Adding event listeners
+let hearts = Array.from(document.getElementsByClassName("heart"));
+hearts.forEach((heart) => {
+  if (heart) {
+    heart.addEventListener("click", on_heart_click);
+  }
+});
+
+/**
+ * @summary This function is a handler for the event of heart-click.
+ * Whenever a user clicks on a heart icon, in case of empty heart:
+ * saves quote in favorites, as well as changing
+ * the heart icon from empty to full.
+ * In case of full heart:
+ * Removes it and switch back to empty heart icon.
+ * Uses the save_or_remove_quote function to handle db operations.
+ */
+function on_heart_click() {
+  const quote = this.parentNode.previousElementSibling.innerHTML.replaceAll(
+    '"',
+    ""
+  );
```

Comment:
Put the quote under a special class/id, and get it using innerText. Don't parse it from inside the HTML